src/features/serverTasks/serverTaskWaiter.ts

```diff
@@ -44,21 +44,26 @@ export class ServerTaskWaiter {
         }, timeout);
 
         while (!stop) {
-            if (pollingCallback) {
-                const taskDetails = await spaceServerTaskRepository.getDetails(serverTaskId);
-                pollingCallback(taskDetails);
+            try {
+                if (pollingCallback) {
+                    const taskDetails = await spaceServerTaskRepository.getDetails(serverTaskId);
+                    pollingCallback(taskDetails);
 
-                if (taskDetails.Task.IsCompleted) {
-                    clearTimeout(t);
-                    return taskDetails.Task;
-                }
-            } else {
-                const task = await spaceServerTaskRepository.getById(serverTaskId);
+                    if (taskDetails.Task.IsCompleted) {
+                        clearTimeout(t);
+                        return taskDetails.Task;
+                    }
+                } else {
+                    const task = await spaceServerTaskRepository.getById(serverTaskId);
 
-                if (task.IsCompleted) {
-                    clearTimeout(t);
-                    return task;
+                    if (task.IsCompleted) {
+                        clearTimeout(t);
+                        return task;
+                    }
                 }
+            } catch (e) {
+                clearTimeout(t);
+                throw e;
```

Comment:
This could be cleaned up with a finally block so we don't need to call clearTimeout before every return.
for example:
```
function DoThings() {
    let stop = false;
    const t = setTimeout(() => {
       stop = true;
    }, 9000);
    try {
        while (!stop) {
            if (true) {
                return "Done!"
            }
         }
     } catch (e) { 
        throw e;
    } finally {
        clearTimeout(t);
    }
}
```
The  finally block gets called before the the return, like wise if we throw an error it will be called before the catch throws is back up. In fact because you are just using the catch to clear the timeout, you can actually remove the catch block all together if you use finally.

Reply:
Excellent pick up. Done.